### PR TITLE
Adding new biocontainers dockerfile

### PR DESCRIPTION
**Pull request by lassefolkersen**
# Submitting a Container
 
 (If you're requesting for a new container, please check the procedure described [here](https://github.com/BioContainers/containers#241-how-to-request-a-container).  
 
 ## Check BioContainers' Dockerfile [specifications](https://github.com/BioContainers/specs)  
 ## Checklist
 
 1. Misc
 - [x] My tool doesn't exist in [BioConda](#make-sure-your-tool-isnt-already-present-in-bioconda)   
 - [x] The image can be built   
 
 2. [Metadata](#check-biocontainers-dockerfile-metadata)  
 - [x] LABEL base_image  
 - [x] LABEL version
 - [x] LABEL software.version  
 - [x] LABEL about.summary  
 - [x] LABEL about.home  
 - [x] LABEL about.license   
 - [x] MAINTAINER  
 
 3. Extra (optionals)
  - [ ] I have written tests in test-cmds.txt  
  - [x] LABEL extra.identifier    
  - [ ] LABEL about.documentation   
  - [x] LABEL about.license_file
  - [x] LABEL about.tags   
 
 
 ## Check [BioContainers'](https://github.com/BioContainers/specs) Dockerfile metadata

### CODE REVIEW

**Comment by lassefolkersen** [comment]
License is LGPL-3.0-only - I'm a bit unsure how to write it correctly though

Where does one edit the biotools entry?

**Comment by osallou** [comment]
if license is not available in SDPX, never mind, warning is only a "warning"  :-)

For bio.tools  it is located at https://bio.tools/  if it exists, you can fill in Dockerfile corresponding label

File: imputeme/v1.0.4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,225 @@
+################## BASE IMAGE ######################
+
+FROM rocker/shiny:4.0.1
+
+
+################## METADATA ######################
+
+LABEL base_image="https://hub.docker.com/repository/docker/lassefolkersen/impute-me"
+LABEL version="v1.0.4"
```
Comment (osallou) [inline imputeme/v1.0.4/Dockerfile]: should be version 1

version matches the Dockerfile version, not the upstream version

File: imputeme/v1.0.4/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+#The impute.me server runs as ubuntu default user, so the docker should too
+RUN useradd ubuntu
+RUN mkdir /home/ubuntu
+RUN chown ubuntu /home/ubuntu && \
+chown ubuntu /var/lib/shiny-server
+USER ubuntu
+WORKDIR /home/ubuntu
+
+#Create the directory stucture
+RUN mkdir /home/ubuntu/srv && \
```
Comment (osallou) [inline imputeme/v1.0.4/Dockerfile:99]: should avoid installing things in /home, with singularity for example, home is mounted over container home, shoumd rather install in/opt for example and use as workdir

**Review by osallou — CHANGES_REQUESTED** [review]

**Comment by lassefolkersen** [comment]
Oh, I see. Thanks a lot for your helpful comments.

It is in bio.tools, actually (https://bio.tools/impute.me). Maybe it's because it's with a . (dot) there. So do you think I should fill in:
`LABEL extra.identifiers.biotools="impute-me,impute.me"
`

The license stuff - I'm going to try with `LGPL-3.0-or-later` instead, which is how it's written in that SPDX list.

I'll change the LABEL version to 
`LABEL version="1.0.0"`
Is that correct? So it's like the version on biocontainers?

That /home mounting, that's a big change, because it's the code from a long-running web-server (running in /home/ubuntu on Amazon EC2 instances - because shouldn't run as root). But I guess I can try to go through the code and change that to something else, what would you suggestion be? /imputeme/<sub-folder?>

**Comment by osallou** [comment]
For version, just set to 1, it is a basic incremebting counter for dockerfile.
For biotools, just set the biotools tool id in label (only one)

File: imputeme/v1.0.4/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,223 @@
+################## BASE IMAGE ######################
+
+FROM rocker/shiny:4.0.1
+
+
+################## METADATA ######################
+
+LABEL base_image="https://hub.docker.com/repository/docker/lassefolkersen/impute-me"
```
Comment (osallou) [inline imputeme/v1.0.4/Dockerfile]: Should just be base image eg lasseflokersen/impute-me

**Review by osallou — CHANGES_REQUESTED** [review]

**Comment by osallou** [comment]
@lassefolkersen merged, thanks for commit and updates

**Pull request merged**